contracts/modules/capital/SafeTracker.sol

```diff
@@ -0,0 +1,152 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.18;
+
+import "@openzeppelin/contracts-v4/token/ERC20/IERC20.sol";
+import "../../abstract/MasterAwareV2.sol";
+import "../../interfaces/IPool.sol";
+import "../../interfaces/IPriceFeedOracle.sol";
+import "../../interfaces/ISafeTracker.sol";
+
+contract SafeTracker is ISafeTracker, MasterAwareV2 {
+
+  uint public coverReInvestmentUSDC;
+
+  string public symbol = "NXMIS";
+  string public name = "NXMIS";
+  uint8 public decimals = 18;
+
+  address public immutable safe;
+  uint public immutable investmentLimit;
+
+  IERC20 public immutable usdc;
+  IERC20 public immutable dai;
+  IERC20 public immutable aweth;
+  IERC20 public immutable debtUsdc;
+
+  /* ========== CONSTRUCTOR ========== */
+
+  constructor(
+    address _master,
+    uint _investmentLimit,
+    address _safe,
+    address _usdc,
+    address _dai,
+    address _aweth,
+    address _debtUsdc
+  ) {
+    require(_master != address(0), "SafeTracker: master cannot be zero address");
+    require(
+      _usdc != address(0) && _dai != address(0) && _aweth != address(0) && _debtUsdc != address(0),
+      "SafeTracker: tokens address cannot be zero address"
+    );
+
+    master = INXMMaster(_master);
```

Comment:
`master` is a storage variable and the contract is a proxy. This will initialize the storage of the implementation and not of the proxy. Remove this parameter from the constructor. See my comment on the issue on notion.